src/legacy/ui/public/chrome/api/__tests__/nav.js

```diff
@@ -79,94 +71,119 @@ describe('chrome nav apis', function () {
-          id: appId,
-          title: 'Discover',
-          linkToLastSubUrl: true,
-          lastSubUrl: `${appUrl}?id=${deletedId}`,
-          url: appUrl
-        }
-      ];
+      fakedLinks = [{
+        id: appId,
+        title: 'Discover',
+        url: `${appUrl}?id=${deletedId}`,
+        appUrl
+      }];
+
       const {
         chrome
-      } = init({ appUrlStore, nav });
+      } = init({
+        appUrlStore,
+        nav: [{
+          id: appId,
+          linkToLastSubUrl: true,
+        }]
+      });
 
       chrome.untrackNavLinksForDeletedSavedObjects([deletedId]);
-      expect(chrome.getNavLinkById('appId').lastSubUrl).to.be(appUrl);
+      expect(coreNavLinks.update.calledWith(appId, { url: appUrl })).to.be(true);
     });
 
     it('should not clear last url when last url does not contains link to deleted saved object', function () {
       const lastUrl = `${appUrl}?id=anotherSavedObjectId`;
       const appUrlStore = new StubBrowserStorage();
-      const nav = [
-        {
-          id: appId,
-          title: 'Discover',
-          linkToLastSubUrl: true,
-          lastSubUrl: lastUrl,
-          url: appUrl
-        }
-      ];
+      fakedLinks = [{
+        id: appId,
+        title: 'Discover',
+        url: lastUrl,
+        appUrl
+      }];
+
       const {
         chrome
-      } = init({ appUrlStore, nav });
+      } = init({
+        appUrlStore,
+        nav: [{
+          id: appId,
+          linkToLastSubUrl: true
+        }]
+      });
 
       chrome.untrackNavLinksForDeletedSavedObjects([deletedId]);
-      expect(chrome.getNavLinkById(appId).lastSubUrl).to.be(lastUrl);
+      expect(coreNavLinks.update.calledWith(appId, { url: appUrl })).to.be(false);
     });
   });
 
   describe('internals.trackPossibleSubUrl()', function () {
     it('injects the globalState of the current url to all links for the same app', function () {
       const appUrlStore = new StubBrowserStorage();
-      const nav = [
+      fakedLinks = [
         {
-          url: 'https://localhost:9200/app/kibana#discover',
-          subUrlBase: 'https://localhost:9200/app/kibana#discover'
+          id: 'kibana:discover',
+          appUrl: 'https://localhost:9200/app/kibana#discover',
         },
         {
-          url: 'https://localhost:9200/app/kibana#visualize',
-          subUrlBase: 'https://localhost:9200/app/kibana#visualize'
+          id: 'kibana:visualize',
+          appUrl: 'https://localhost:9200/app/kibana#visualize',
         },
         {
-          url: 'https://localhost:9200/app/kibana#dashboards',
-          subUrlBase: 'https://localhost:9200/app/kibana#dashboard'
+          id: 'kibana:dashboard',
+          appUrl: 'https://localhost:9200/app/kibana#dashboards',
         },
-      ].map(l => {
-        l.lastSubUrl = l.url;
-        return l;
-      });
+      ];
 
       const {
         internals
-      } = init({ appUrlStore, nav });
+      } = init({
+        appUrlStore,
+        nav: [
```

Comment:
This is enforced by prettier which is not applied to js files. I'll add go ahead and add the trailing commas.

Reply:
Good catch, and thanks for the info!

Reply:
These do not have trailing commas either, so we may want to check the linter settings to see if there are other problems.